src/app/lib/atlasmap-data-mapper/services/mapping.management.service.ts

```diff
@@ -91,35 +88,29 @@ export class MappingManagementService {
     public fetchMappings(mappingFileNames: string[], mappingDefinition: MappingDefinition): Observable<boolean> {
         return new Observable<boolean>((observer:any) => {
             if (mappingFileNames.length == 0) {
-                console.log("No pre-existing mapping exists.");
                 observer.complete();
                 return;
             }
-            var startTime: number = Date.now();
 
-            var baseURL: string = this.cfg.initCfg.baseMappingServiceUrl + "mapping/";
-            var operations: any[] = [];
-            for (let mappingName of mappingFileNames) {
-                var url: string = baseURL + mappingName;
+            const baseURL: string = this.cfg.initCfg.baseMappingServiceUrl + "mapping/";
+            const operations: any[] = [];
+            for (const mappingName of mappingFileNames) {
+                const url: string = baseURL + mappingName;
                 DataMapperUtil.debugLogJSON(null, "Mapping Service Request", this.cfg.initCfg.debugMappingServiceCalls, url);
-                let operation = this.http.get(url).map((res:Response) => res.json());
+                const operation = this.http.get(url).map((res:Response) => res.json());
                 operations.push(operation);
             }
             Observable.forkJoin(operations).subscribe((data:any[]) => {
```

Comment:
Well, I'm not that familiar with this service and its overall purpose, so for me it's a bit hard to understand all its nuances, but please bear in mind that this subscription will remain open even when the service is disposed by the Angular dep injection system (which is usually at the end of the application lifespan, so it's not a big deal after all). If that is okay for you, then go ahead.

Otherwise, you can always desubscribe from it after certain condition applies. Usually it is not a good idea to leave RxJS subscriptions open in the wild, but this is a service and services are meant to persist, including its logic, so I leave this call up to you ;)

Reply:
Filed an issue for this one, I think we need to check if there're same issues elsewhere in the code and fix them:
https://github.com/atlasmap/atlasmap-ui/issues/130